articles/contributing/style-guide.md

```diff
@@ -0,0 +1,157 @@
+---
+title: Q# Style Guide | Microsoft Docs
+description: Q# Style Guide
+author: cgranade
+ms.author: chgranad
+ms.date: 10/12/2018
+ms.topic: article
+uid: microsoft.quantum.contributing.style
+---
+
+# Q# Style Guide #
+## General Conventions ##
+
+- The conventions listed here are suggestions only, and should likely be disregarded when they would result in less readable or useful code.
+  Put differently, disregarding conventions should always be an intentional decision to produce more useful code, and not an accident.
+
+## Naming Conventions ##
+
+In offering the Quantum Development Kit, we strive for function and operation names that help quantum developers write programs that are easy to read and that minimize surprise.
+An important part of that is that when we choose names for functions, operations, and types, we are establishing the *volcabulary* that programmers use to express quantum concepts; with our choices, we either help or hinder them in their effort to clearly communicate.
```

Comment:
```suggestion
An important part of that is that when we choose names for functions, operations, and types, we are establishing the *vocabulary* that programmers use to express quantum concepts; with our choices, we either help or hinder them in their effort to clearly communicate.
```

Reply:
Unless this was an intentional decision? :-)

Reply:
Nope, thanks for the fix.